src/realm/dictionary.cpp

```diff
@@ -740,6 +740,12 @@ void Dictionary::clear()
         if (recurse)
             _impl::TableFriend::remove_recursive(*m_obj.get_table(), cascade_state); // Throws
     }
+    if (size() == 0) {
```

Comment:
this is always true right?
To follow the behaviour of other collections, we'd only emit a clear if there was anything to clear, so you can move this into the conditional above and use the replication object from there.

Reply:
I think we need to keep this at the end of the function, because of recursion. However, the `if(size() == 0)` is redundant.